### PR TITLE
feat: cms marketplace admin

### PR DESCRIPTION
**Pull request by bashleigh**
# Pull request checklist

**Detail as per issue below (required):**

fixes: #7227

### CODE REVIEW

File: packages/app-marketplace-admin/src/components/apps-browse/app-browse-table.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+} from '@reapit/elements'
+import { AppsBrowseConfigEnum, AppsBrowseConfigItemInterface } from '@reapit/foundations-ts-definitions'
+import { GetActionNames, getActions } from '@reapit/utils-common'
+import { useReapitUpdate } from '@reapit/utils-react'
+import React, { FC } from 'react'
+import { colorSquare } from './app-browse.styles'
+import { reapitConnectBrowserSession } from '../../core/connect-session'
+import { shleemy } from 'shleemy'
+
+export const AppBrowseTable: FC<{
```
Comment (willmcvay) [inline packages/app-marketplace-admin/src/components/apps-browse/app-browse-table.tsx:20]: I see you've re-used the AppMarket code but can you re-name the components to something that makes sense please? Also, probably needs a tests?

File: packages/app-marketplace-admin/src/components/apps-browse/app-browse-upsert-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+    })
+
+    setLoading(false)
+    if (returned) {
+      upsertItem(returned as AppsBrowseConfigItemInterface)
+      closeModal()
+    }
+  }
+
+export const AppBrowseUpsertModal: FC<{
```
Comment (willmcvay) [inline packages/app-marketplace-admin/src/components/apps-browse/app-browse-upsert-modal.tsx:90]: Can you add a few tests please

File: packages/app-marketplace-admin/src/components/apps-browse/app-browse-upsert.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import { Loader, PageContainer, Title, useModal } from '@reapit/elements'
+import { AppsBrowseConfigEnum, AppsBrowseConfigItemInterface } from '@reapit/foundations-ts-definitions'
+import { GetActionNames, getActions } from '@reapit/utils-common'
+import { useReapitGet } from '@reapit/utils-react'
+import React, { FC, useEffect, useState } from 'react'
+import { reapitConnectBrowserSession } from '../../core/connect-session'
+import { AppBrowseTable } from './app-browse-table'
+import { AppBrowseUpsertModal } from './app-browse-upsert-modal'
+
+export const AppBrowseUpsert: FC<{}> = () => {
```
Comment (willmcvay) [inline packages/app-marketplace-admin/src/components/apps-browse/app-browse-upsert.tsx:11]: And again, could use a test or two

**Review by willmcvay — CHANGES_REQUESTED** [review]

File: packages/app-marketplace-admin/src/components/apps-detail/__tests__/apps-detail.test.tsx [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+import { render } from '../../../tests/react-testing'
+import { AppsDetail } from '../apps-detail'
+
+describe('AppsDetail', () => {
```
[the remark this answers is not in the document]
Comment (bashleigh) [reply packages/app-marketplace-admin/src/components/apps-detail/__tests__/apps-detail.test.tsx:5]: ahhh didn't see them

**Review by willmcvay — APPROVED** [review]

**Pull request merged**